src/platforms/ios/InternetChecker.swift

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import NetworkExtension
+
+class InternetChecker {
+    private let logger = IOSLoggerImpl(tag: "InternetChecker")
+    private let numberOfPings = 1
+    private let pingTimeLimit: Double = 0.1 // 100 milliseconds
```

Comment:
I am a little concerned that this is a little too aggressive and inflexible. IIUC, this is sending a ping to the wireguard server's public IPv4 address to verify internet connectivity and 100ms is not going to be enough to reach a server on the other side of the internet.

Reply:
Discussed in a meeting this morning; I'll add some more context in code comments..